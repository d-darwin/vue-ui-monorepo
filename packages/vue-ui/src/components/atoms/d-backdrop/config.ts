import { CssPosition } from "./types";
import styles from "./index.css?module";

export default {
  name: "DBackdrop",
  class: styles.dBackdrop,
  opacity: 0.5,
  zIndex: 1000,
  position: CssPosition.fixed,
} as const;
