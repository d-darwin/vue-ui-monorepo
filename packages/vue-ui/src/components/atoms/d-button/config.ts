export default {
  name: "DButton",
  className: "dButton",
};
