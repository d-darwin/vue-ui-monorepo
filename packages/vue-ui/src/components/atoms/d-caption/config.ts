import styles from "./index.css?module";

export default {
  name: "DCaption",
  class: styles["dCaption"],
} as const;
