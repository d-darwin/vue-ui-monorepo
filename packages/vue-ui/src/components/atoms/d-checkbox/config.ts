export default {
  name: "DCheckbox",
  className: "dCheckbox",
  inputClassName: "input",
  labelClassName: "label",
  labelInnerClassName: "labelInner",
  errorClassName: "error",
  iconContainerClassName: "iconContainer",
  iconContainerBackdropClassName: "iconContainerBackdrop",
  iconClassName: "icon",
  checkMark: "âœ“",
};
