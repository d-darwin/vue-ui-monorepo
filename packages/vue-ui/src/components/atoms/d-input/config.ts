export default {
  name: "DInput",
  className: "dInput",
};
