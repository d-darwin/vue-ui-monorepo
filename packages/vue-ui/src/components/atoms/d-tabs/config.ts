export default {
  tabsName: "DTabs",
  tabName: "DTab",
  tabsClassName: "dTabs",
  tabClassName: "dTab",
};
