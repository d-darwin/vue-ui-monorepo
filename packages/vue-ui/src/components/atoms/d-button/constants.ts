import type { DLoaderProps } from "@darwin-studio/vue-ui/src/components/atoms/d-loader/types";
import config from "./config";
import styles from "./index.css?module";

export const LOADER_DEFAULTS: DLoaderProps = {
  class: [styles[config.loaderClassName]],
};
