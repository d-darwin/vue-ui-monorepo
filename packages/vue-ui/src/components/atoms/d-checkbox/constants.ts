import { DCaptionProps } from "@darwin-studio/vue-ui/src/components/atoms/d-caption/types";
import { TYPE } from "@darwin-studio/vue-ui/src/components/atoms/d-caption/constant";
import config from "@darwin-studio/vue-ui/src/components/atoms/d-switch/config";
import styles from "./index.css?module";

export const DEFAULT_VALUE = "on";

export const CAPTION_DEFAULTS: DCaptionProps = {
  type: TYPE.DANGER,
  class: styles[config.captionClassName],
};
