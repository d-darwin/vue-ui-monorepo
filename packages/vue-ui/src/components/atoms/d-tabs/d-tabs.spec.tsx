import { mount, shallowMount } from "@vue/test-utils";
import { baseClassCase } from "@darwin-studio/vue-ui/src/utils/test-case-factories";
import { SIZE } from "@darwin-studio/vue-ui-codegen/dist/constants/size";
import { FONT } from "@darwin-studio/vue-ui-codegen/dist/constants/font";
import { DTabs, DTab, DTabpanel } from "./index";
import config from "./config";

// TODO: move to the config
jest.mock("chalk", () => ({
  greenBright: jest.fn(),
  red: jest.fn(),
  yellowBright: jest.fn(),
}));

describe("DTabs", () => {
  const wrapper = mount(DTabs);

  /*  baseClassCase(wrapper, config.tabsClassName);

  it("Should render config.tablistClassName to the tablist's class name", () => {
    const tablist = wrapper.find(`.${config.tablistClassName}`);
    expect(tablist).toBeTruthy();
  });

  it("Should render props.tablistLabel to the tablist's aria-label attr", async () => {
    const tablistLabel = "Some aria label";
    await wrapper.setProps({ tablistLabel });
    const tablist = wrapper.find(`.${config.tablistClassName}`);
    expect(tablist.attributes("aria-label")).toBe(tablistLabel);
  });

  it("Should props.tablistClass to the tablist's class name", async () => {
    const tablistClass = "some-tablist-class";
    await wrapper.setProps({ tablistClass });
    const tablist = wrapper.find(`.${config.tablistClassName}`);
    expect(tablist.classes()).toContain(tablistClass);
  });
*/
  it("Should render props.tabs as content of the tablist", async () => {
    // TODO: without mounting new component ???
    const wrapper = await mount(DTabs, {
      props: {
        tabs: [
          <DTab id="111" tabpanelId="111-1" active={true} label={"Tab 1"} />,
          <DTab id="222" tabpanelId="222-2" label={"Tab 2"} />,
        ],
      },
    });

    const tablist = wrapper.find(`.${config.tablistClassName}`);
    const tabs = tablist.findAllComponents(DTab);
    expect(tabs?.[0]?.element).toMatchSnapshot(); // TODO: find out other way to compare
    expect(tabs?.[1]?.element).toMatchSnapshot(); // TODO: find out other way to compare
  });

  it("Should render slots.tabs as content of the tablist", async () => {
    // TODO: without mounting new component ???
    const wrapper = await mount(DTabs, {
      slots: {
        tabs: [
          <DTab id="111" tabpanelId="111-1" active={true} label={"Tab 1"} />,
          <DTab id="222" tabpanelId="222-2" label={"Tab 2"} />,
        ],
      },
    });

    const tablist = wrapper.find(`.${config.tablistClassName}`);
    const tabs = tablist.findAllComponents(DTab);
    expect(tabs?.[0]?.element).toMatchSnapshot(); // TODO: find out other way to compare
    expect(tabs?.[1]?.element).toMatchSnapshot(); // TODO: find out other way to compare
  });

  /*
    // TODO: props.tabsSize
    it("Should pass props.tabSize as props.size to containing DTab components", async () => {
      const tabSize = SIZE.LARGE;
      await wrapper.setProps({
        tabSize,
        tabs: [<DTab id="111" active={true} label={"Tab 1"} />],
      });
      const tabWrapper = wrapper.findComponent(DTab);
      console.log(tabWrapper.classes());
      expect(tabWrapper.classes()).toBe(SIZE.LARGE);
    });

    // TODO: DTab id autogenerate
    it("Should pass autogenerated id to containing DTab components if not passed", async () => {
      expect(true).toBeFalsy();
    });

    // TODO: DTab <-> DTabpanel connection
    it("Should pass DTabpanel's prop.id to the corresponding DTab as prop.tabpanelId", async () => {
      expect(true).toBeFalsy();
    });

    // TODO: props.tabpanels
    it("Should render props.tabpanels as content the list of tabpanels", async () => {
      const tabpanel1 = shallowMount(DTabpanel);
      const tabpanel2 = shallowMount(DTabpanel);
      await wrapper.setProps({ tabs: [tabpanel1, tabpanel2] });
      expect(true).toBeFalsy();
    });

    // TODO: slots.tabpanels
    it("Should render slots.tabs as content of the tablist", async () => {
      const tabpanel1 = shallowMount(DTabpanel);
      const tabpanel2 = shallowMount(DTabpanel);
      const wrapper = await mount(DTabs, {
        slots: {
          tabs: [tabpanel1, tabpanel2],
        },
      });
      expect(true).toBeFalsy();
    });

    // TODO: props.tabpanelsFont
    it("Should pass props.tabpanelsFont as props.font to containing DTabpanel components", async () => {
      const tabpanelsFont = FONT.SMALL;
      const tabpanel = await shallowMount(DTabpanel);
      await wrapper.setProps({ tabpanelsFont, tabs: [tabpanel] });
      expect(true).toBeFalsy();
    });

    // TODO: DTabpanel id autogenerate
    it("Should pass autogenerated id to containing DTabpanel components if not passed", async () => {
      expect(true).toBeFalsy();
    });

    // TODO: DTabpanel <-> DTab connection
    it("Should pass DTab's prop.id to the corresponding DTabpanel as prop.tabId", async () => {
      expect(true).toBeFalsy();
    });*/

  // Classes and behaviour
  // TODO: props.disabled
  // TODO: props.padding
  // TODO: props.transition
  // TODO: props.tag
  // TODO: props.tablistTag
  // TODO: props.enableHtml

  // TODO: DTab.props
  // active ???
  // TODO: DTab.onClick
  // TODO: DTab.whenClick
  // TODO: DTabpanel.props
  // active ???

  // TODO: arrow navigation
});
