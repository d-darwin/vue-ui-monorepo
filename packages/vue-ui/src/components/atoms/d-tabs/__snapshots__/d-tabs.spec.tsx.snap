// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DTabs Should render props.tabs as content of the tablist 1`] = `
<li
  aria-controls="111-1"
  aria-selected="true"
  class="dTab fontMedium paddingDefaultMedium paddingDefault outlinePrimaryMedium sizeMedium transitionFast active"
  id="111"
  role="tab"
  tabindex="0"
>
  Tab 1
</li>
`;

exports[`DTabs Should render props.tabs as content of the tablist 2`] = `
<li
  aria-controls="222-2"
  class="dTab fontMedium paddingDefaultMedium paddingDefault outlinePrimaryMedium sizeMedium transitionFast"
  id="222"
  role="tab"
  tabindex="-1"
>
  Tab 2
</li>
`;

exports[`DTabs Should render slots.tabs as content of the tablist 1`] = `
<li
  aria-controls="111-1"
  aria-selected="true"
  class="dTab fontMedium paddingDefaultMedium paddingDefault outlinePrimaryMedium sizeMedium transitionFast active"
  id="111"
  role="tab"
  tabindex="0"
>
  Tab 1
</li>
`;

exports[`DTabs Should render slots.tabs as content of the tablist 2`] = `
<li
  aria-controls="222-2"
  class="dTab fontMedium paddingDefaultMedium paddingDefault outlinePrimaryMedium sizeMedium transitionFast"
  id="222"
  role="tab"
  tabindex="-1"
>
  Tab 2
</li>
`;
